src/sources/splunk_hec.rs

```diff
@@ -246,18 +247,20 @@ impl SplunkSource {
     }
 
     fn health_service(&self) -> BoxedFilter<(Response,)> {
-        let credentials = self.credentials.clone();
+        let valid_credentials = self.valid_credentials.clone();
         let authorize =
             warp::header::optional("Authorization").and_then(move |token: Option<String>| {
-                let credentials = credentials.clone();
+                let valid_credentials = valid_credentials.clone();
                 async move {
-                    match (token, credentials) {
-                        (_, None) => Ok(()),
-                        (Some(token), Some(password)) if token.as_bytes() == password.as_ref() => {
-                            Ok(())
+                    if valid_credentials.is_empty() {
+                        return Ok(());
+                    }
+                    if let Some(token) = token {
+                        if valid_credentials.contains(&token) {
+                            return Ok(());
                         }
-                        _ => Err(Rejection::from(ApiError::BadRequest)),
                     }
+                    return Err(Rejection::from(ApiError::BadRequest));
```

Comment:
Nit I think this could probably be rewritten a bit to look something like:

```
match token {
  Some(token) if valid_credentials.contains(&token) => Ok(()),
  _ => Err(Rejection::from(ApiError::BadRequest))
}
```